src/blister.c

```diff
@@ -1155,9 +1128,7 @@ UInt SizeBlist (
     /* loop over the blocks, adding the number of bits of each one         */
     n = 0;
     for ( i = 1; i <= nrb; i++ ) {
-        m = *ptr++;
-        COUNT_TRUES_BLOCK(m);
-        n += m;
+        n += COUNT_TRUES_BLOCK(*ptr++);
```

Comment:
Here you could try manual loop unrolling:
```C
#if USE_POPCNT && defined(HAVE___BUILTIN_POPCOUNTL)
    while (nrb > 0) {
        UInt n1 += COUNT_TRUES_BLOCK(*ptr++);
        UInt n2 += COUNT_TRUES_BLOCK(*ptr++);
        UInt n3 += COUNT_TRUES_BLOCK(*ptr++);
        UInt n4 += COUNT_TRUES_BLOCK(*ptr++);
        n += n1+n2+n3+4;
        nrb -= 4;
    }
#endif
    for ( i = 1; i <= nrb; i++ ) {
        n += COUNT_TRUES_BLOCK(*ptr++);
    }
```

Reply:
nrb > 3 rather than nrb > 0?

Reply:
Does the manual unrolling achieve something? I think manual unrolling is rather discouraged.

Reply:
Did you read http://danluu.com/assembly-intrinsics/ ?